product-service/src/products.ts

```diff
@@ -0,0 +1,91 @@
+export interface Product {
+    count: number;
+    description: string;
+    id: string;
+    price: number;
+    title: string;
+}
+
+export async function getProducts(): Promise<Product[]> {
+    return new Promise((resolve) => {
+        resolve(products);
+    });
+}
+
+export async function getProductById(productId: string): Promise<Product> {
+    return new Promise<Product>((resolve, reject) => {
+        if (!productId) {
+            reject('Empty productId');
+            return;
+        }
+        const product = products.find(pr => pr.id === productId);
+        if (product) {
+            resolve(product);
+        } else {
+            reject(`Product with id ${productId} not found`);
+        }
+    })
+}
+
+const products: Product[] = [
```

Comment:
I assume it would be better to store products in separate file. But this will change in next hw, so no big deal.

Reply:
Yep, it's a temporary implementation